qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -415,10 +358,13 @@ def status(self) -> JobStatus:
             return self._status
 
         with api_to_job_error():
-            api_response = self._api_client.job_status(self.job_id())
+            api_response = self._runtime_client.job_get(self.job_id())["state"]
+            # response state possibly has two values: status and reason
+            # reason is not used in the current interface
             self._api_status = api_response["status"]
+            api_metadata = self._runtime_client.job_metadata(self.job_id())
             self._status, self._queue_info = self._get_status_position(
-                self._api_status, api_response.get("info_queue", None)
+                self._api_status, api_metadata
```

Comment:
Do we really need to call `job_metadata()` here? The old code updated the queue info because the IQX API also returned queue info as part of status. Since Runtime API doesn't do that, we can call `job_metadata()` lazily.